### ISSUE TITLE
batch-norm cannot support unrolled rnn input data

### ISSUE THREAD

**Issue by quanzongfeng** [apache/incubator-mxnet#3110]
For unrolled rnn/lstm,  we unroll the input sequence.  When i tried to use batch-norm for input data after embed layer, i  fount it impossible for all characters in a sequnce with just one batch-norm.
first, batch-norm calc mean and var in forward function every invoking, just store in output and used in backward function; but in unrolled lstm, for one sequence with length m, it may invoke forward function m times, and we only get the last invoke mean and var which store in output, so we loss some mean and var
second, moving_mean and moving_var are two ndarrays which we can not assign when the batch_norm operator defined, the two ndarrays are defined when the batch_norm op created, which is not support copy.  for sequence with m length in rnn/lstm, there may m times aux_states for batch_norm. 

Is there any good way to use batch-norm in unrolled rnn/lstm?

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
